File: 0007-reverse-integer/0007-reverse-integer.cpp

class Solution {
public:
    long Reverse(long x,int len)
    {
        if(len<=1)
            return x;
        
        int a= 0; int b = 0;
        int r = len/2;
        int s = len - len/2;
        int m=round(pow(10,r));
        int M = round(pow(10,s));

        a = x/m;
        b = x%m;
        x = (long)((long)(Reverse(b,r)*M) + (long)Reverse(a,s));
        return x;
    }
    int reverse(int x) {
        ios_base::sync_with_stdio(0);
        int len=0;
        long y;

            y =x;
        while(y!=0)
        {
            y = y/10;
            len++;
        }
        y = x;
        y = Reverse(y,len);
        if(y>pow(2,31)-1||y<-pow(2,31)) return 0;

        return y;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        isNeg = False
        arr = []
        b = ""
        if len(str(x)) == 1:
            return x
        while x % 10 == 0:
            x = int(x / 10)
        if x < 0:
            isNeg = True
            x *= -1
        for i in str(x):
            arr.append(i)
        for j in arr[::-1]:
            b += j
        c = int(b)
        if isNeg:
            c *= -1
        if c.bit_length() > 31:
            return 0
        else:
            return c
